lib/servers/SockJSServer.js

```diff
@@ -57,8 +57,14 @@ module.exports = class SockJSServer extends BaseServer {
     connection.close();
   }
 
-  // f should return the resulting connection
+  // f should return the resulting connection and, optionally, the connection headers
   onConnection(f) {
-    this.socket.on('connection', f);
+    this.socket.on('connection', (connection) => {
+      f(connection, connection.headers);
```

Comment:
Why not use only `connection`?

Reply:
@evilebottnawi Because `ws` connection object does not work the same, it does not have a `headers` property (https://github.com/webpack/webpack-dev-server/pull/2077/files#diff-b111fb9e1bc06a00edb8d0416cabf86dR33). The goal is to not assume anything about the `connection` object, since it could differ between socket server types.